lib/puppet/file_serving/terminus_helper.rb

```diff
@@ -29,4 +29,16 @@ def path2instances(request, *paths)
       path2instance(request, base_path, :ignore_source_permissions => true, :relative_path => file)
     end
   end
+
+  def self.unescape_url(url)
+    url.sub(/^url=/,'')
+  end
+
+  def self.escape_url(url)
+    if url =~ /^https?:/
+      "url=#{url}"
+    else
+      url
+    end
```

Comment:
Adding a comment from the PR triage. Since puppet's source parameter already supports other URL types, e.g. file, puppet, it would ideally not be necessary to wrap/unwrap http as a special case. It would be good to understand where puppet goes haywire when given an http URL.